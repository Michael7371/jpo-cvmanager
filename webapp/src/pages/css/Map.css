.rsu-marker {
  margin-top: -6px;
  margin-left: -6px;
}

.legend {
  z-index: 90;
  background: #0e2052;
  margin: 10px;
  border-radius: 15px;
}

.legend-header {
  font-family: Arial, Helvetica, sans-serif;
  font-size: 24px;
  margin-top: 5px;
  margin-left: 10px;
  margin-right: 10px;
  color: white;
}

.legend-item {
  margin-top: 8px;
  margin-bottom: 8px;
  margin-left: 10px;
  margin-right: 10px;
}

.legend-label {
  font-family: Arial, Helvetica, sans-serif;
  font-size: 18px;
  color: white;
}

.legend-input {
  margin-right: 10px;
  accent-color: #d16d15;
}

.legend-grid {
  position: absolute;
}

.rsu-status-div {
  z-index: 90;
  background: #0e2052;
  border-radius: 15px;
  margin-top: -29px;
}

.rsu-status-label {
  display: block;
  font-family: Arial, Helvetica, sans-serif;
  font-size: 18px;
  color: white;
  margin-top: 8px;
  margin-bottom: 8px;
  margin-left: 10px;
  margin-right: 10px;
}

.rsu-status-input {
  margin-right: 10px;
  accent-color: #d16d15;
}

.map-button {
  height: 35px;
  padding: 1px 12px;
  margin-top: -120px;
  margin-left: 10px;
  text-align: center;
  text-decoration: none;
  font-size: 18px;
  background: #d16d15;
  border-radius: 30px;
  border: none;
  cursor: pointer;
  color: white;
  z-index: 90;
}
