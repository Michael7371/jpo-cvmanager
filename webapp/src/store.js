import { configureStore } from '@reduxjs/toolkit'
import rsuReducer from './generalSlices/rsuSlice'
import userReducer from './generalSlices/userSlice'
import wzdxReducer from './generalSlices/wzdxSlice'
import configReducer from './generalSlices/configSlice'
import adminAddOrganizationReducer from './features/adminAddOrganization/adminAddOrganizationSlice'
import adminAddRsuReducer from './features/adminAddRsu/adminAddRsuSlice'
import adminAddUserReducer from './features/adminAdduser/adminAddUserSlice'
import adminEditOrganizationReducer from './features/adminEditOrganization/adminEditOrganizationSlice'
import adminEditRsuReducer from './features/adminEditRsu/adminEditRsuSlice'
import adminEditUserReducer from './features/adminEditUser/adminEditUserSlice'
import adminOrganizationTabReducer from './features/adminOrganizationTab/adminOrganizationTabSlice'
import adminOrganizationTabUserReducer from './features/adminOrganizationTabUser/adminOrganizationTabUserSlice'
import adminOrganizationTabRsuReducer from './features/adminOrganizationTabRsu/adminOrganizationTabRsuSlice'
import adminRsuTabReducer from './features/adminRsuTab/adminRsuTabSlice'
import adminUserTabReducer from './features/adminUserTab/adminUserTabSlice'
import menuReducer from './features/menu/menuSlice'
import rsuUpdateMenuReducer from './features/rsuUpdateMenu/rsuUpdateMenuSlice'

export const setupStore = (preloadedState) => {
  return configureStore({
    reducer: {
      rsu: rsuReducer,
      user: userReducer,
      wzdx: wzdxReducer,
      config: configReducer,
      adminAddOrganization: adminAddOrganizationReducer,
      adminAddRsu: adminAddRsuReducer,
      adminAddUser: adminAddUserReducer,
      adminEditOrganization: adminEditOrganizationReducer,
      adminEditRsu: adminEditRsuReducer,
      adminEditUser: adminEditUserReducer,
      adminOrganizationTab: adminOrganizationTabReducer,
      adminOrganizationTabUser: adminOrganizationTabUserReducer,
      adminOrganizationTabRsu: adminOrganizationTabRsuReducer,
      adminRsuTab: adminRsuTabReducer,
      adminUserTab: adminUserTabReducer,
      menu: menuReducer,
      rsuUpdateMenu: rsuUpdateMenuReducer,
    },
    preloadedState,
  })
}
