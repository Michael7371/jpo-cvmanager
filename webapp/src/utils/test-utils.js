/**
 * This function searches for the every somewhat random/chaotic class name and property that cause snapshot tests to be inconsistent.
 * This current list includes MUI classes (css-*) and aria-invalid attributes.
 *
 * @ex :
 * ```
 * const { container } = render(<Component />);
 *
 * replaceChaoticIds(container);
 * ```
 *
 * @param container The HTMLElement node to search for SSR ids
 */
function replaceChaoticIds(container) {
  const props = [
    {
      selector: 'class',
      updateFunc: (val) => val.replace(/css-[0-9a-z]*?-/g, 'css-mocked-'),
    },
  ]

  container.querySelectorAll('*').forEach((item) => {
    props.forEach((prop) => {
      if (item.getAttribute(prop.selector)) {
        item.setAttribute(prop.selector, prop.updateFunc(item.getAttribute(prop.selector)))
      }
    })
  })
  container.querySelectorAll('input[aria-invalid]').forEach((item) => {
    item.removeAttribute('aria-invalid')
  })
  return container
}

export { replaceChaoticIds }
