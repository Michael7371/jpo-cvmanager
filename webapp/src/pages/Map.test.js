import React from 'react'
import { render } from '@testing-library/react'
import Map from './Map'
import { Provider } from 'react-redux'
import { setupStore } from '../store'
import { replaceChaoticIds } from '../utils/test-utils'

it('should take a snapshot', () => {
  const initialState = {
    rsu: {
      value: {
        bsmCoordinates: [],
        rsuCounts: {},
        mapList: [],
      },
    },
  }
  const { container } = render(
    <Provider store={setupStore(initialState)}>
      <Map />
    </Provider>
  )

  expect(replaceChaoticIds(container)).toMatchSnapshot()
})
