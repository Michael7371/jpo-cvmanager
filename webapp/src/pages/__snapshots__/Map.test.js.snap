// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="MuiGrid-root legend-grid MuiGrid-container MuiGrid-align-items-xs-center"
    >
      <div
        class="legend"
      >
        <h1
          class="legend-header"
        >
          Map Layers
        </h1>
        <div
          class="legend-item"
        >
          <label
            class="legend-label"
          >
            <input
              checked=""
              class="legend-input"
              type="checkbox"
            />
            RSU
          </label>
        </div>
        <div
          class="legend-item"
        >
          <label
            class="legend-label"
          >
            <input
              class="legend-input"
              type="checkbox"
            />
            Heatmap
          </label>
        </div>
        <div
          class="legend-item"
        >
          <label
            class="legend-label"
          >
            <input
              class="legend-input"
              type="checkbox"
            />
            BSM Viewer
          </label>
        </div>
        <div
          class="legend-item"
        >
          <label
            class="legend-label"
          >
            <input
              class="legend-input"
              type="checkbox"
            />
            WZDx
          </label>
        </div>
      </div>
      <div
        class="rsu-status-div"
      >
        <h1
          class="legend-header"
        >
          RSU Status
        </h1>
        <label
          class="rsu-status-label"
        >
          <input
            checked=""
            class="rsu-status-input"
            name="none-status-radio"
            type="radio"
            value="none"
          />
          None
        </label>
        <label
          class="rsu-status-label"
        >
          <input
            class="rsu-status-input"
            name="online-status-radio"
            type="radio"
            value="online"
          />
          Online Status
        </label>
        <label
          class="rsu-status-label"
        >
          <input
            class="rsu-status-input"
            name="scms-status-radio"
            type="radio"
            value="scms"
          />
          SCMS Status
        </label>
      </div>
    </div>
    <div
      class="container-fluid"
      style="width: 100%; height: calc(100vh - 100px); display: flex;"
    >
      <div
        style="position: relative; width: 100%; height: 100%;"
      />
    </div>
  </div>
</div>
`;
